audformat/core/utils.py

```diff
@@ -936,9 +947,17 @@ def join_schemes(
           labels: [a, b]
 
     """
-    labels = join_labels([db.schemes[scheme_id].labels for db in dbs])
-    for db in dbs:
-        db.schemes[scheme_id].replace_labels(labels)
+    if all([isinstance(db.schemes[scheme_id].labels, str) for db in dbs]):
+        tables = [db[db.schemes[scheme_id].labels] for db in dbs]
+        index = union([table.index for table in tables])
```

Comment:
Here we only the index of the tables, but maybe we also want to concatenate the columns?

Reply:
On the other hand, this is something that `Database.update()` will take care of. So I guess it's enough to combine the index here.

Reply:
Good point, so in the tests we should not only create misc tables with an index, but with columns as well (and most likely columns that use a scheme).

Reply:
I added those tests in #276 with `Database.update()`. If we agree that `join_schemes()` will only update the index I don' t think we need to extend the tests here.

Reply:
But it will be strange if we don't join the columns. This means we can end up with two databases that have the same scheme (as stated in the docstring of `join_schemes()`), but when using `.get(map=)` it might fail for one of the databases or return different results.

Reply:
But it also means that we have to repeat a lot of checks from `Database.update()` like if all columns have the same dtype etc. I would rather add a note in the docstring that it will not updated the columns and the user should call `Database.update()`.

Reply:
Also check for rater stuff, etc.

Reply:
The main reason we have `join_schemes()` is that `Database.update()` does not support updating schemes, see https://github.com/audeering/audformat/issues/61.
So, I guess we would then need to add the missing functionality to `Table.update()`.

The other question is why would I ever call `join_schemes()` if it does not correctly update the scheme of the databases?

Reply:
> I would rather add a note in the docstring that it will not updated the columns

Before doing that I would then not support misc tables in `join_schemes()`, extend `Database.update()` and `Table.update()` to support non matching schemes, and state in the error message of `join_schemes()` that the user should use `Database.update()` for misc tables.

My original idea of updating `join_schemes()` was, that we can then reuse it inside `Table.update()` to add support for non matching schemes and updating misc tables, but if you think this can be better or only implemented directly inside `audformat.core.table.Base.update()` then I would vote for not supporting misc tables in `join_schemes()`.

Reply:
Currently `Database.update()` fails when the index of the misc tables that are used in a scheme do not match. The can be fixed by calling `join_schemes()`. Then `Database.update()` will take care of also updating the columns. So I think the combination of `join_schemes()` + `Database.update()` might work and #276 should now pass.

I mean, I see your point that it would be nice if `join_schemes()` also directly updates the columns. But it's not easy to achieve, e.g. if the misc table of one database has an additional column that links to a scheme that is not in the other database we have to copy that scheme. In `Database.update()` we do such things, but I'm not too motivated to implement the same kind of logic also in `join_schemes()`. It's also more complicated since in `Database.update()` we only have to update the database we call from, but in `join_schemes()` we have to apply it to all databases that are passed.

If you want open an issue, but I would prefer not to work in this PR and first see if #276 will also work with the current solution. Then we at least offer a way to update misc tables schemes. If we later change `join_schemes()` it should still work.

Reply:
> Before doing that I would then not support misc tables in `join_schemes()`, extend `Database.update()` and `Table.update()` to support non matching schemes, and state in the error message of `join_schemes()` that the user should use `Database.update()` for misc tables.

Ok that is of course also a solution. I will try to implement it directly in #276

Reply:
OK, thanks for the detailed explanation.

One alternative would be to implement updating the index inside `Database.update()` and not adding support for misc tables to `join_schemes()`. Then we could also later add this functionality.

But I guess we need `join_schemes()` already in `Database.update()` for the other schemes (non misc tables schemes) and it would make it easier if it supports misc tables as well, correct?

Reply:
I wasn't fast enough, you will try already ;)